apps/docs/pages/guides/platform/exhaust-ram.mdx

```diff
@@ -0,0 +1,53 @@
+
+## Issues with High Memory Usage
+
+Every Supabase project runs in its dedicated virtual machine. Your instance will have a different set of hardware provisioned depending on your [compute add-on](https://supabase.com/docs/guides/platform/compute-add-ons). Your hardware may not be suitable for the intended workload and may experience high RAM usage.
+
+High RAM usage could come with a range of issues:
+
+- degraded performance overall as your instance might have to rely more on swap
+- the operating system might start killing processes as your system runs out of memory
+- in rare cases, your instance may become unresponsive
+
+## Monitor your RAM
+
+You can check your RAM usage on the Supabase Platform. For this go to database health in the reports section or [click here](https://app.supabase.com/project/_/reports/database) and select your project.
+
+<Admonition type="note">
+
+High memory usage doesn't necessarily mean that your instance is at risk. Memory that is used for caching and buffers improves data access speed. But if you notice less performance alongside high memory usage, your memory usage might be unhealthy.
+
+</Admonition>
+
+It is also possible to monitor your resources and set up alerts using Prometheus/Grafana. With Grafana you will be able to see how much of your RAM is used for caching. Moreover, you can track other metrics such as your Swap usage. You can find a guide for this [here](https://supabase.com/docs/guides/platform/metrics).
+
+## Common reasons for high RAM usage
+
+Everything you do with your Supabase project requires memory in some form. Hence, there can be many reasons for high RAM usage. Here are some common ones:
+
+- **Query performance:** Queries that take a long time to complete (>1 second) could be using your RAM inefficiently. Check our guide on [examining query performance](https://supabase.com/docs/guides/platform/performance#examining-query-performance).
+- **Too many connections:** Every connection to your database consumes memory. You can check the number of active connections [here](https://supabase.com/dashboard/project/_/database/roles) after you select your project. We also have a separate guide on too many connections [here](https://supabase.com/docs/guides/platform/troubleshooting#too-many-open-connections).
+- **Extensions:** Some extensions such as timescale or pg_cron can use a lot of memory. It can also add up when you have too many extensions running.
+
+## How to fix your memory issues
+
+1. **Upgrade your compute:** You can get a Compute Add-on for your project. Follow [this link](https://app.supabase.com/project/_/settings/billing/update/pro) (follow [this link](https://app.supabase.com/project/_/settings/billing/update/team) if for the Team plan and this link for the Enterprise plan) and select your project to see your upgrade options.
```

Comment:
I just made a change that it will automatically redirect to the correct plan (https://github.com/supabase/supabase/pull/15036)

So you can just link to  https://app.supabase.com/project/_/settings/billing/update/pro
```suggestion
1. **Upgrade your compute:** You can get a Compute Add-on for your project. Follow [this link](https://app.supabase.com/project/_/settings/billing/update/pro) and select your project to see your upgrade options.
```

Reply:
ah great thanks!